### DISCUSSION TITLE
Support for plain date device class

### DISCUSSION THREAD

**Discussion by scop** [home-assistant/architecture#610] [Entity Models] [answered]
I'm working on a local waste pickup service integration. The collection schedule from them has a resolution of date, i.e. they tell which date the pickup is going to be (with some leeway but let's ignore that here), there is no specific time of day for it.

Specifying the sensor as `DEVICE_CLASS_TIMESTAMP` mostly works, but it works as if the time of pickup would be at midnight UTC. This results in somewhat confusing and incorrect information in the UI when it starts to say e.g. that the pickup will be "in 2 hours" or was "13 minutes ago" when the collection hasn't yet happened today, and the correct thing for the sensor to say would be "today" for as long as the current date is the same as the scheduled pickup date.

Any thoughts how to handle this? Add `DEVICE_CLASS_DATE`? What if we come across something whose resolution is a week, or a month, or an hour, etc? Maybe a range of some kind?

(I guess JS doesn't have a date-only object (the Date is really a timestamp), so the UI would have to be told by some other means that it should ignore hours/minutes/seconds for values like these depending on case, or to use something else besides Date.)

**Comment by balloob** [comment]
I think `DEVICE_CLASS_DATE` makes sense. 

Or should this be a calendar platform ?

**Reply by frenck** [reply]
I think a state class makes sense. For example, the garbage gets picked up 18 august 2021 (2021-08-18, not 2021-08-18T00:00:00+0000)
IMHO, it would be cool if it could display it in the frontend relatively as well (just like we do with `DEVICE_CLASS_TIMESTAMP`). E.g., `today`, `tomorrow`, `yesterday`.

This would be useful for things like indeed, garbage collection, birthdays, and all other stuff that is date bound (not time).

**Reply by dgomes** [reply]
it is relevant for billing too (energy, water, gas), "when does my bill gets calculated ?"

**Comment by MartinHjelmare** [comment, marked as answer]
Implemented in https://github.com/home-assistant/core/pull/55887.
